lib/rouge/lexer.rb

```diff
@@ -501,12 +506,45 @@ def self.detect?(text)
   end
 
   module Lexers
+    ProxyLexer = Struct.new(:path, :name, :tag, :aliases, :filenames, :mimetypes) do
+      def method_missing(name, *args, &blk)
+        Lexers.load_lexer self.path
+        lexer = Lexers.const_get self.name
+        lexer.send(name, *args, &blk)
+      end
+    end
+
     @_loaded_lexers = {}
+    @_loaded_proxies = {}
 
     def self.load_lexer(relpath)
       return if @_loaded_lexers.key?(relpath)
       @_loaded_lexers[relpath] = true
       load File.join(__dir__, 'lexers', relpath)
     end
+
+    def self.load_proxies(cache_file)
+      registry = {}
+
+      JSON.parse(File.read(cache_file)).each do |h|
+        proxy = ProxyLexer.new(h["path"],
+                               h["name"],
+                               h["tag"],
+                               h["aliases"] || [],
+                               h["filenames"] || [],
+                               h["mimetypes"] || [])
+        @_loaded_proxies[proxy.name] = proxy
+        registry[proxy.tag] = proxy
+        proxy.aliases.each { |a| registry[a] = proxy } unless proxy.aliases.nil?
```

Comment:
Short-circuit with `&&`
```suggestion
        proxy.aliases && proxy.aliases.each { |a| registry[a] = proxy }
```
That way, once we drop support for EOL Rubies in the future, this can be simplified to use the safe nav operator from Ruby 2.3:
```ruby
proxy.aliases&.each { |a| registry[a] = proxy }
```

Reply:
Thanks for the catch—this is an oversight based on how ProxyLexer's `aliases` argument previously worked. It's now always an array so the entire check can be removed.